Modules/_elementtree.c

```diff
@@ -3669,7 +3669,7 @@ _elementtree_XMLParser___init___impl(XMLParserObject *self, PyObject *target,
                            (unsigned long)_Py_HashSecret.expat.hashsalt);
     }
 
-    if (target) {
+    if (target && target != Py_None) {
```

Comment:
Replace `target: object = NULL` with `target: object = None` above and simplify this check:
```suggestion
    if (target != Py_None) {
```